This patch makes it possible to reboot mi by mqtt and web.

chefpro opened the pull request:
In my setup with battery instead of solarpanel and a shelly switching mains, it is needed to restart the mi after setting mains to on.
If I not doing so, mi needs forever to enable.

Restart dtu is not enough to solve this.

This patch makes it possible to reboot mi by mqtt and web.

ohAnd commented:
to understand the use case:
- instead of pv panel the supply of ~ 50V will be given by a battery
- instead of controlling the input dc voltage you switching the main of the inverter to the grid
- means inverter/ dtu is running always, but the grid voltage is pulled off for the inverter to don't produce energy to the grid

is that right?

in my setup I also used battery and I've switched for under voltage use cases the battery supply off (so that the mi/dtu has no supply) - and producing (amount) or not to the grid was given by power limit and inverter on/off - after coming back with DC (after under voltage) the inverter started again in the right status

in summary:
to fulfill the PR it will make sense that there is some explanation of the feature in readme incl. a short description for which uses cases it is needed/ useful

chefpro commented:
> to understand the use case:
> 
>     * instead of pv panel the supply of ~ 50V will be given by a battery

Yes, it is a 16s lifepo4.

> 
>     * instead of controlling the input dc voltage you switching the main of the inverter to the grid
> 

Yes, I also thought to switch the DC side, but a cheap DC Relais is hard to find. It also needs to solder a ESP32 Board myself. A 12€ Shelly is then a better and faster solution in my opinion.

>     * means inverter/ dtu is running always, but the grid voltage is pulled off for the inverter to don't produce energy to the grid
> 
> 
> is that right?

Yes, thats right.

> 
> in my setup I also used battery and I've switched for under voltage use cases the battery supply off (so that the mi/dtu has no supply) - and producing (amount) or not to the grid was given by power limit and inverter on/off - after coming back with DC (after under voltage) the inverter started again in the right status

In my setup I nearly never have under voltage. I just cut of when sun go down and switch on when sun rises.

> 
> in summary: to fulfill the PR it will make sense that there is some explanation of the feature in readme incl. a short description for which uses cases it is needed/ useful

Is this really a matter of specific use cases? I think there are plenty. I would more see that as a step forward to feature complete. All times the Micro Inverter has a problem a reboot should be possible. Not everybody has the advance of relais, but switch of by hand. There are much more Api in the proto files which can be of some use, but not usable by mqtt for now. In this case it is also a feature that break nothing. It's just making proto Api reachable by mqtt and web.

I agree that some sort of documentation is a good idea, so I added that in the readme.

ohAnd commented:
found some minor things, see review comments

generally:
- webui: the single "prominent" button ... - I will think about it ;-) - maybe you have other ideas, too
- mqtt integration: to be aligned with overall integration - HA AutoDiscovery Config is missing for the reboot setter -> but this could be a single issue (feature enhancement) for later integration

chefpro commented:
> found some minor things, see review comments
> 
> generally:
> 
>     * webui: the single "prominent" button ... - I will think about it ;-) - maybe you have other ideas, too

Yes, I tried to make it smaller, but I am not very much into this html css stuff. Could you assist please ? :-)
Maybe some menu or expert function popup. I don't care that much. I am more a function c/c++ guy, not a good designer. :-D

> 
>     * mqtt integration: to be aligned with overall integration - HA AutoDiscovery Config is missing for the reboot setter -> but this could be a single issue (feature enhancement) for later integration

Yes, I am more into fhem. So, I do not use HA. Could you assist with that, too?

Thanks, Peter

PS: I can not find any review comments in the code?

chefpro commented:
PPS: I also added more documentation as mentioned.

ohAnd commented:
ah sorry, I used the review comment function... 
for readme:
=> ... and the extension of the web ui ("clickable reboot of mi possible") is maybe also useful
for dtuInterface.cpp:
=> line 136, 141, 1172 -> minor: doubled with dtu restart - should be something like "DTUinterface:\t ... " ...Device => ...Mi

found during testing with device and serial logging...

chefpro commented:
![grafik](https://github.com/user-attachments/assets/756ff7e5-2643-447c-a421-d9bc1ebe7592)
I testet it.
There is something wrong with the overlay.
Should setting hide when reboot is clicked?

ohAnd commented:
> ![grafik](https://private-user-images.githubusercontent.com/2660040/450627012-756ff7e5-2643-447c-a421-d9bc1ebe7592.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg5Mzc3NTgsIm5iZiI6MTc0ODkzNzQ1OCwicGF0aCI6Ii8yNjYwMDQwLzQ1MDYyNzAxMi03NTZmZjdlNS0yNjQzLTQ0N2MtYTQyMS1kOWJjMWViZTc1OTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDNUMDc1NzM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmU1NTFjZThiYTc4Nzk4ZTFhOWIxNDVkYzJjNTE0MjZjYWRmZmM0Mjk4OTIxN2QxMzAwNDBhYmNkZTllNTBhNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.eJXqfrrmKG2aiMqhbMgCOwHO3vjQY2XxxQ66o5hj6X4) I testet it. There is something wrong with the overlay. Should setting hide when reboot is clicked?

that's the point if the browser cache is not refreshed ... means the old style.css is then still used - clearing the cache will show the right...

Heini1985 commented:
Hallo ohAnd,

Ich habe die Buttons noch obwohl ich die neueste Firmware auf esp32 gglasht habe. Den Cache habe ich im browser auch scho gelöscht.
![image](https://github.com/user-attachments/assets/c740f3b2-74ff-46d5-afe5-a73e06a33dd0)

Wie bekommen ich die Button bei mir in die Einstellungen?

Viele Grüße
